### ISSUE TITLE
Strange handling of BG class in `BBoxHead`

### ISSUE THREAD

**Issue by gf0507033** [open-mmlab/mmdetection#6192]
I can't figure out the following code in `BBoxHead`. According to [`regress_by_class`](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/models/roi_heads/bbox_heads/bbox_head.py#L481-L483):
```python
            label = label * 4
            inds = torch.stack((label, label + 1, label + 2, label + 3), 1)
            bbox_pred = torch.gather(bbox_pred, 1, inds)
```
As per [docs](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/models/roi_heads/bbox_heads/bbox_head.py#L388), width of `bbox_pred` is `4 * num_classes`

According to [these lines](), it's assumed that `label` mostly generated by  following [code](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/models/roi_heads/bbox_heads/bbox_head.py#L159-L164):
```
        # original implementation uses new_zeros since BG are set to be 0
        # now use empty & fill because BG cat_id = num_classes,
        # FG cat_id = [0, num_classes-1]
        labels = pos_bboxes.new_full((num_samples, ),
                                     self.num_classes,
                                     dtype=torch.long)
```
All values of `labels` except first `num_pos` are `num_classes`. So in `regress_by_class` max value of `label` will be `4 * num_classes + 3`. How do we correctly gather corresponding values from `bbox_pred` with width `4 * num_classes`?

I can't run any HTC model with semantic info because of this misunderstanding :(

**Comment by AronLin** [comment]
The positive labels are from `0` to `num_classes-1`. So the max value of `label` will be `4*num_classes-1`

**Comment by gf0507033** [comment]
@AronLin once again:
- `labels` is filled by `num_classes` value
- some (not all) first positions of `labels` are replaced by another values
- max value in `inds = torch.stack((4 * label, 4 * label + 1, 4 * label + 2, 4 * label + 3), 1)` is `num_classes + 1`

Any model with `reg_class_agnostic=False` in `BBoxHead` is broken now. Any models in mmdetection's model zoo uses  `reg_class_agnostic=True`.

**Comment by AronLin** [comment]
When we set the BG to the label, we have not used the function `regress_by_class`.
So if you want to use this function, please ensure that the background label is filtered.
like:
![image](https://user-images.githubusercontent.com/29054582/135403654-4bc4cc00-df21-44ba-8089-67b0a7af57f6.png)
